File: src/main/java/org/kane/domain/DTO/entityDTO/weight_record/for_chart/WeightChartDataDTO.java
package org.kane.domain.DTO.entityDTO.weight_record.for_chart;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.kane.database.entity.physical_quantity.HumanWeight;
import org.kane.database.enum_types.ChartPeriodType;

import java.time.LocalDate;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class WeightChartDataDTO {
    private List<WeightPointDTO> weightList;
    private ChartPeriodType period;
    private HumanWeight minWeight;
    private HumanWeight maxWeight;
    private HumanWeight avgWeight;
    private HumanWeight weightChange;
    private LocalDate startDate;
    private LocalDate endDate;
}

File: src/app/DTO/entity_dto/diary/weight_record/weight-chart-data.dto.ts
import WeightPointDTO from './weight-point.dto';
import {ChartPeriodType} from '../../../types';

export default interface WeightChartDataDTO{
  weightList: WeightPointDTO[];
  period: ChartPeriodType;
  minWeight: number;
  maxWeight: number;
  avgWeight: number;
  weightChange: number;
  startDate: Date;
  endDate: Date;
}
